packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -161,6 +165,88 @@ export class DemoModePromptGatherer implements ParametersGatherer<{}> {
   }
 }
 
+export class SelectLwcComponentDir
+  implements ParametersGatherer<{ fileName: string; outputdir: string }> {
+  private typeDir: string;
+  private typeDirRequired: boolean | undefined;
+  public static readonly defaultOutput = path.join('main', 'default');
+
+  public constructor(typeDir: string, typeDirRequired?: boolean) {
+    this.typeDir = typeDir;
+    this.typeDirRequired = typeDirRequired;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string; outputdir: string }>
+  > {
+    let packageDirs: string[] = [];
+    try {
+      packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    } catch (e) {
+      if (
+        e.name !== 'NoPackageDirectoryPathsFound' &&
+        e.name !== 'NoPackageDirectoriesFound'
+      ) {
+        throw e;
+      }
+    }
+
+    const dirOptions = this.getDefaultOptions(packageDirs);
+    let outputdir = await this.showMenu(dirOptions);
+    const namePathMap = new Map();
+    let filepath;
+    let filename;
+    let fileName;
+
+    if (outputdir) {
+      const fullPath = path.join(getRootWorkspacePath(), outputdir);
+      const registry = new RegistryAccess();
+      const components = registry.getComponentsFromPath(fullPath);
+      const lwccomponents = components.filter(
+        lwc =>
+          lwc.type.name === registryData.types.lightningcomponentbundle.name
+      );
+      const lwcNames = lwccomponents.map(lwc => lwc.fullName);
+
+      for (const component of lwccomponents) {
+        namePathMap.set(component.fullName, component.xml);
+      }
```

Comment:
We can simplify this a bit by filtering, building the map, and getting the names with one iteration of the components:

```typescript
const lwcNames = [];
for (const component of components) {
  const { fullName, type } = component;
  if (type.name === registryData.types.lightningcomponentbundle.name) {
    namePathMap.set(fullName, component.xml);
    lwcNames.push(fullName);
  }
}
```

Or you can get the lwcNames just from the keys of the map `namePathMap.keys()`. This returns an iterator though so you might have to do `Array.from(namePathMap.keys())`

Reply:
This looks much better! Thanks!